project/src/main/java/com/google/sps/data/PlacesFetcher.java

```diff
@@ -0,0 +1,157 @@
+import com.google.maps.model.PlaceDetails;
+import com.google.maps.model.PlaceType;
+import com.google.maps.model.PlacesSearchResponse;
+import com.google.maps.model.PriceLevel;
+import com.google.maps.GeoApiContext;
+import com.google.maps.TextSearchRequest;
+import com.google.maps.PlaceDetailsRequest;
+import com.google.maps.errors.ApiException;
+import com.google.maps.model.PlacesSearchResult;
+import com.google.maps.PlacesApi;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.google.appengine.repackaged.com.google.common.collect.ImmutableList;
+
+public class PlacesFetcher {
+
+    /** Places will be fetched within a ceratain radius from this location. */
+    private final LatLng location;
+
+    /** The cuisine types of the places that are fetched. */
+    private final String cuisineType;
+
+    /** The maximum price level as identified in Google Places of the fetched places. */
+    private final PriceLevel maxPriceLevel;
+
+    /** Specifies if the fetched places must be open at the time of fetching. */
+    private final boolean openNow;
+
+    /** The type of places that will be searched. */
+    private static final PlaceType TYPE = PlaceType.RESTAURANT;
+
+    /** The search radius for place. */
+    private static final int SEARCH_RADIUS = 5000;
+    // TODO (M1): check at least 10 results, and if less extend radius
+
+    /** The entry point for a Google GEO API request. */
+    private static final GeoApiContext CONTEXT = new GeoApiContext.Builder()
+        .apiKey("AIza...") // TODO : save key in a file that can be accessed and pushed to github
+        .build();
+
+    /**
+     * Fields are temporaraly hard coded for M0 version.
+     * In next versions those same fields will be the fields of a UserPrefrences
+     * instance passed to the PlacesFetcher constructor by the Servlet.
+     */
+    public PlacesFetcher() {
+        this.location = new LatLng(32.080576, 34.780641); // Rabin Square TLV
+        this.cuisineType = "sushi"; // TODO (talbarnahor): change to set of types
+        this.maxPriceLevel = PriceLevel.values()[2];
+        this.openNow = true;
+    }
+
+    // TODO (talbarnahor): add exception handling and testing
+    /**
+     * Builds a query and requests it from Google Places API.
+     *
+     * @return list of places that supply the query.
+     * @throws IOException
+     * @throws InterruptedException
+     * @throws ApiException
+     */
+    public List<Place> fetch() throws IOException, InterruptedException, ApiException {
+        PlacesSearchResult[] results = getPlacesSearchResults();
+        return createPlacesList(results);
+    }
+
+    /**
+     * Queries Google Places API according to given params.
+     *
+     * @return A PlacesSearchResponse which contains the search results
+     * @throws ApiException
+     * @throws InterruptedException
+     * @throws IOException
+     */
+    public PlacesSearchResult[] getPlacesSearchResults()
```

Comment:
I changed it to public because otherwise I wasn't able to stub it in the tester. 
Anyway I will try to rearrange it like you suggested on the testing comment and maybe it will solve this visibility issue.

Reply:
OK, let me know when you finish the test change.